tools/commissioner_thci/commissioner_impl.py

```diff
@@ -416,6 +418,7 @@ def _expect(self, expected, timeout=10):
                 return
 
             self._sleep(0.1)
+            timeout -= 0.1
```

Comment:
Would the `sleep` add unnecessary delay to the THCI? I think a better model is to only sleep when there is not a line to read yet. 
I would suggest several changes to the current code:
- `_read_line()` should only return `None` when there isn't a complete line (ends with \n) to return. I think the current implementation returns None in unnecessary conditions. 
- The code that calls `_read_line()` should not sleep if `_read_line()` successfully returns a line, but only sleep if `_read_line()` returns `None`
- Can use a smaller sleep interval: e.x. 0.01 for lower delay.

Reply:
@Moandor-y Thoughts about returning `None`?

Reply:
`_read_line` will return one unnecessary `None` for each new line. To solve that we can change this function to:
```python
    def _read_line(self):
        logging.debug('Reading line')
        if len(self._lines) > 1:
            return CONTROL_SEQUENCE.sub('', self._lines.pop(0))

        tail = ''
        if len(self._lines):
            tail = self._lines.pop()

        try:
            tail += self._read()
        except socket.error:
            logging.debug('No new data')

        self._lines += NEW_LINE.split(tail)

        if len(self._lines) > 1:
            return CONTROL_SEQUENCE.sub('', self._lines.pop(0))
```